### DISCUSSION TITLE
Should the red blinks be there?

### DISCUSSION THREAD

**Discussion by rogerthn2019** [UtilitechAS/amsreader-firmware#732] [General]
Hi
I did happen to see red blink on my Pow-K+ as on [https://www.youtube.com/shorts/YlMpvfMDF2Q](url)
Seems to be working OK but I needed to ask ;-)
![image](https://github.com/UtilitechAS/amsreader-firmware/assets/57598997/ea9a5447-f23a-4712-a213-d600af245af2)

**Comment by ArnieO** [comment]
Interesting!
No, there should be no red blink. Normally, there should be a red indicator or an error message shown if there are is red blinking.

As indicated in the [Firmware v2.2. User Manual](https://www.amsleser.no/index.php?controller=attachment&id_attachment=53), a single red blink indicates that the meter received invalid data:
![image](https://github.com/UtilitechAS/amsreader-firmware/assets/10295178/2e0ce399-38d3-438f-a03e-9576d8c55d81)

I propose you take a peek "under the hood" by running a Telnet debug: https://www.amsleser.no/module/ets_blog/blog?id_post=24

**Comment by rogerthn2019** [comment]
I did get `Device is booting, please wait` when disabling debug and now the red blinks are gone.
Maybe uptime (52 days) "overflow"?
Enclosing telnet log
[telnet.log](https://github.com/UtilitechAS/amsreader-firmware/files/14319228/telnet.log)
There are some 
`No HAN data received last 30s, single blink`

**Reply by ArnieO** [reply]
Interesting. 
It was probably the reboot that fixed it, but the problem should not have been there in the first case. On Pow-K there is no cable issue, so it should not fail the reading in this way. 

Uptime 52 days should certainly not give overflow. 

Which version Pow-K hardware are you running? A screendump of your Info-page should provide the information I need to assess.

**Comment by rogerthn2019** [comment]
Reboot is known to fix a lot but shouldn't be needed?
![image](https://github.com/UtilitechAS/amsreader-firmware/assets/57598997/1b704656-2b83-4329-bf17-e3f81cc3c7de)

**Reply by ArnieO** [reply]
OK, so this is a Pow-K that you bought from us? 
If you still have the issue of "No HAN data received last 30s" after reboot (that should not be needed), I suspect an issue with your board - and would like to replace it.
Please email us at post@amsleser.no.

**Comment by rogerthn2019** [comment]
Yes, Pow-K+ bought from you.
It has been replaced once, due to unexpected reboots by you 2023-03-30.
There are no "No HAN data received last 30s" right now.
Let's wait and see if it happen again, maybe after 49.7 days uptime.

**Reply by ArnieO** [reply]
OK, that is a good plan!

**Comment by gskjold** [comment]
Found the bug, uptime (millis) rollover is the issue

**Comment by rogerthn2019** [comment]
I did happen to see red blinks again before the upgrade to v2.3.6, uptime +50 days.
Enclosing v2.3.5 debug from telnet
[telnet.log](https://github.com/user-attachments/files/15926888/telnet.log)

**Reply by ArnieO** [reply]
Huh...  "No HAN data received last 30s, single blink".

@gskjold : The uptime rollover issue should have been fixed? https://github.com/UtilitechAS/amsreader-firmware/discussions/732#discussioncomment-8627217
